src/main/java/com/conveyal/r5/analyst/Grid.java

```diff
@@ -126,96 +127,154 @@ public Grid (int zoom, int width, int height, int north, int west) {
         this.grid = new double[width][height];
     }
 
+    public static class PixelWeight {
+        final int x;
+        final int y;
+        final double weight;
+
+        private PixelWeight (int x, int y, double weight){
+            this.x = x;
+            this.y = y;
+            this.weight = weight;
+        }
+    }
+
+    /**
+     * A functional interface (callback interface) that processes one pixel weight at a time.
+     * This avoids storing massive lists of intermediate results that have been known to eat memory on the server.
+     * This happens when processing geometries that contain a huge amount of pixels, which a user might include in
+     * a shapefile.
+     */
+    public interface PixelWeightCallback {
```

Comment:
If most of the speedup comes from use of PreparedGeometry, and these in-memory pixel weights are not one of our major memory consumers in this application, we should probably just keep the programming interface for this pixel weighting code as simple as possible, i.e. just do them all in memory since many of them need to be in memory.